tensorflow/core/kernels/mkl/mkl_quantize_op.cc

```diff
@@ -509,12 +484,19 @@ class MklQuantizeV2Op : public OpKernel {
     AllocateOutputSetMklShape(ctx, 2, &output_max_tensor, max_tf_shape,
                               max_mkl_shape, native_format);
 
+    // Create the oneDNN wrapper over Eigen threadpool and set max threads
+    // in oneDNN.
+    Eigen::ThreadPoolInterface* eigen_interface =
+        EigenThreadPoolFromTfContext(ctx);
+    tsl::OneDnnThreadPool eigen_tp(eigen_interface,
+                                   ThreadPoolUseCallerThread());
+
     float scale_factor = 0;
     if (mode_ == QUANTIZE_MODE_SCALED) {
       // Estimating scales for quantization.
       const int num_bits = sizeof(T) * 8;
       const float max_abs = std::max(std::abs(min_range), std::abs(max_range));
-      const bool is_signed = std::is_signed<T>::value;
+      const bool is_signed = std::is_same<T, qint8>();
```

Comment:
Why only qint8?  What about for other signed quantized types T?

If the issue is that: 

> I found that when T = qint8
> 
> Windows: std::is_signed<T>::value is false
> Linux: std::is_signed<T>::value is true

then _that_ should be fixed.

Reply:
> Why only qint8? What about for other signed quantized types T?
> 
> If the issue is that:
> 
> > I found that when T = qint8
> > Windows: std::is_signed::value is false
> > Linux: std::is_signed::value is true
> 
> then _that_ should be fixed.

I have added other quantized types as well in the last commit.

Regarding the earlier issue, it was with compiler (MSVC at the time Google CI test failed earlier). I am not sure if Google has moved to Clang or not. Therefore, I am not sure if compiler needs to be fixed or not.

Reply:
Closing the loop on this: `std::is_signed` is supposed to return `false`, since these are custom types, so we are not allowed to override `std::is_arithmetic` or `std::is_signed` on them (doing so results in UB).  It should return `false` with both clang and MSVC (and does for me).

Google uses clang internally for everything.  But our GitHub CI for Windows still using MSVC.